packages/pigeon/lib/pigeon_lib.dart

```diff
@@ -274,18 +274,23 @@ abstract class Generator {
   void generate(StringSink sink, PigeonOptions options, Root root);
 }
 
+DartOptions _dartOptionsWithCopyrightHeader(
+    DartOptions? dartOptions, String? copyrightHeader) {
```

Comment:
It is really weird from an API understandability perspective that these options are both nullable.

Please strongly consider making `options.dartOptions` non-nullable in a follow-up so that you don't have to have a helper whose sole purpose is to merge something into an object, which doesn't actually require that the object being merged into exists.